docs/documentation/ui_customization/topics/localization.adoc

```diff
@@ -0,0 +1,96 @@
+[[_localization]]
+== Internationalization and Localization
+
+Keycloak supports localized text throughout all user interfaces.
+
+.Prerequisites
+
+* To enable internationalization for a realm, you must enable internationalization in the `Realm settings` section of the admin console. See the {adminguide_link}[{adminguide_name}].
+
+=== Localizing mesages in a theme
+
+Text in the templates is loaded from message bundles. A theme that extends another theme will inherit all messages from the parent's message bundle and you can
+override individual messages by adding `<THEME TYPE>/messages/messages_en.properties` to your theme.
+
+For example to replace `Username` on the login form with `Your Username` for the `mytheme` create the file
+`themes/mytheme/login/messages/messages_en.properties` with the following content:
+
+[source]
+----
+usernameOrEmail=Your Username
+----
+
+Within a message values like `{0}` and `{1}` are replaced with arguments when the message is used. For example {0} in `Log in to {0}` is replaced with the name
+of the realm.
+
+Texts of these message bundles can be overwritten by realm-specific values. The realm-specific values are manageable via UI and API.
+
+=== Adding a language to a theme
+
+
+
+.Procedure
+
+. Create the file `<THEME TYPE>/messages/messages_<LOCALE>.properties` in the directory of your theme.
+
+. Add this file to the `locales` property in `<THEME TYPE>/theme.properties`.
+For a language to be available to users the realms `login`, `account` and `email`, the theme has to support the language, so you need to add your language for those theme types.
```

Comment:
```suggestion
. Add this file to the `locales` property in `<THEME TYPE>/theme.properties`.
+
For a language to be available to users and the `login`, `account` and `email` realms, the theme has to support the language, so you need to add your language for those theme types.
```

Reply:
> Rewording is my best guess.

Those are not realms.  They are theme types.  But it's a little confusing because as I recall, they are just referred to as `themes`.

This language comes from the current documentation, so I think it's safe to leave it alone for now.

Reply:
Okay, not sure what it should be, but it seems there is missing punctuation somewhere.  
I was thinking, it should be "For a language to be available to users COMMA..." but I not sure.